buildspecs/pr-build-amzn.yml

```diff
@@ -53,20 +54,29 @@ phases:
 
       # Define the Go version to install
       - GOVERSION="$(cat GO_VERSION)"
-      
+
       # Install Go and define variables based on Amazon Linux version (2 or 2023)
       # Amazon Linux 2023 uses package manager DNF, while older versions use YUM
-      # Set the appropriate AL23 version echo string to include in build log
+      # First, try to install the specified Go version, if unable, install latest Go version available
       - |
         if [[ "$amzn_version" = "amzn2023" ]]; then
           sudo dnf --releasever="$AL23_VERSION" update -y
-          sudo yum install -y golang-$GOVERSION
-          amzn_version="amzn2023"
-          al23_version_echo="Amazon Linux 2023 Version: $AL23_VERSION"
+          echo "Amazon Linux 2023 Version: $AL23_VERSION" 2>&1 | tee -a $BUILD_LOG
+          if sudo dnf install -y golang-$GOVERSION; then
```

Comment:
Why don't we use `install-golang` ([Makefile target](https://github.com/aws/amazon-ecs-agent/blob/master/Makefile#L371-L372) or the [script](https://github.com/aws/amazon-ecs-agent/blob/master/scripts/install-golang.sh) itself) to download/install the desired version of go?

I think we want CodeBuild builds to use the specified version of go because that version of go will be used to build the artifacts actually released to customers.